drivers/gpio/gpio_brcmstb.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2024 Junho Lee <junho@tsnlab.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT brcm_brcmstb_gpio
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/irq.h>
+
+#define GIO_ODEN  0x00
+#define GIO_DATA  0x04
+#define GIO_IODIR 0x08
+#define GIO_EC    0x0c
+#define GIO_EI    0x10
+#define GIO_MASK  0x14
+#define GIO_LEVEL 0x18
+#define GIO_STAT  0x1c
+
+#define SET_BIT(reg, pin) reg |= (uint32_t)(1 << pin);
+#define SET_BIT_RW(reg, addr, pin)                                                                 \
+	do {                                                                                       \
+		reg = sys_read32(addr);                                                            \
+		SET_BIT(reg, pin);                                                                 \
+		sys_write32(reg, addr);                                                            \
+	} while (0)
+#define CLR_BIT(reg, pin) reg &= ~((uint32_t)(1 << pin));
+#define CLR_BIT_RW(reg, addr, pin)                                                                 \
+	do {                                                                                       \
+		reg = sys_read32(addr);                                                            \
+		CLR_BIT(reg, pin);                                                                 \
+		sys_write32(reg, addr);                                                            \
+	} while (0)
+
+#define DEV_CFG(dev)  ((const struct gpio_brcmstb_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct gpio_brcmstb_data *)(dev)->data)
```

Comment:
Do not use `DEV_*` macros to access data and config.

Reply:
All uses of those macros have been removed. Should I also remove them? If so, using the `DEVICE_MMIO_NAMED_MAP` macro is inconvenient.